docs/reference/aggregation.md

```diff
@@ -0,0 +1,488 @@
+The provided buckets have to be continous. During the aggregation, the values extracted from the fast_field field will be checked against each bucket range.
+Note that this aggregation includes the from value and excludes the to value for each range.
+
+```json
+{
+    "my_ranges": {
+        "field": "score",
+        "ranges": [
+            { "to": 3.0 },
+            { "from": 3.0, "to": 7.0 },
+            { "from": 7.0, "to": 20.0 },
+            { "from": 20.0 }
+        ]
+    }
+}
+```
+
+#### Limitations/Compatibility
+
+Overlapping ranges are not yet supported.
+
+The keyed parameter (elasticsearch) is not yet supported.
+
+
+#### Parameters
+
+###### **field**
+
+The field to aggregate on.
+
+###### **ranges**
+
+The list of buckets, with from and to values. 
+The from value is inclusive in the range.
+The to value is not inclusive in the range.
+
+The first bucket can omit the `from` value, and the last bucket the `to` value.
+Note that this aggregation includes the from value and excludes the to value for each range. Extra buckets will be created until the first to, and last from, if necessary.
+
+
+### Terms
+
+Creates a bucket for every unique term.
+
+```json
+{
+    "genres": {
+        "terms":{ "field": "genre" }
+    }
+}
+```
+
+
+#### Terminology
+Shard parameters are supposed to be equivalent to elasticsearch shard parameter.
+Since they are
+
+#### Document count error
+To improve performance, results from one segment are cut off at `segment_size`. On a index with
+a single segment this is fine. When combining results of multiple segments, terms that
+don't make it in the top n of a shard increase the theoretical upper bound error by lowest
+term-count.
+
+Even with a larger `segment_size` value, doc_count values for a terms aggregation may be
+approximate. As a result, any sub-aggregations on the terms aggregation may also be approximate.
+`sum_other_doc_count` is the number of documents that didn’t make it into the the top size
+terms. If this is greater than 0, you can be sure that the terms agg had to throw away some
+buckets, either because they didn’t fit into size on the root node or they didn’t fit into
+`segment_size` on the segment node.
+
+#### Per bucket document count error
+If you set the `show_term_doc_count_error` parameter to true, the terms aggregation will include
+doc_count_error_upper_bound, which is an upper bound to the error on the doc_count returned by
+each segment. It’s the sum of the size of the largest bucket on each segment that didn’t fit
+into segment_size.
+
+#### Parameters
+
+###### **field**
+
+The field to aggregate on.
+
+###### **size**
+
+By default, the top 10 terms with the most documents are returned. Larger values for size are more expensive.
+
+###### **segment_size**
+
+
+The get more accurate results, we fetch more than size from each segment.
+Increasing this value is will increase the cost for more accuracy.
```

Comment:
"Increasing this value is will increase the accuracy but also the CPU/memory usage"?